src/libcollections/btree/node.rs

```diff
@@ -454,38 +417,28 @@ impl<K, V> NodeRef<marker::Owned, K, V, marker::Internal> {
     /// Similar to `ascend`, gets a reference to a node's parent node, but also
     /// deallocate the current node in the process. This is unsafe because the
     /// current node will still be accessible despite being deallocated.
-    pub unsafe fn deallocate_and_ascend(self) -> Option<
-        Handle<
-            NodeRef<
-                marker::Owned,
-                K, V,
-                marker::Internal
-            >,
-            marker::Edge
-        >
-    > {
+    pub unsafe fn deallocate_and_ascend
+        (self)
+         -> Option<Handle<NodeRef<marker::Owned, K, V, marker::Internal>, marker::Edge>> {
```

Comment:
The `self` argument on its own here in particular looks weird. I also think that the split open type is more clear, but that's more debatable.